### PR TITLE
Updated UI

### PR DESCRIPTION
**Pull request by mrinalwahal**
Following tasks are done:

1. Python `gogs_client` is removed and auth is shifted to a bare API call.
2. UI of the front-end has been updated.
3. Docker image updated to have pre-installed snakemake so to avoid installing snakemake in every build job.
4. Code is cleaner than before.
5. PEP8 styling is enabled and py-linting is being adhered to.
6. You can no longer access the workflow page on the front-end without logging in first.
7. Front-end packages are reduced and only the required ones are kept.

### CODE REVIEW

File: back-end/service.py [hunk 1]
```diff
@@ -1,172 +1,301 @@
...
+            )
+        public_key = key.public_key().public_bytes(
+            serialization.Encoding.OpenSSH,
+            serialization.PublicFormat.OpenSSH
+        )
+
+        os.makedirs(SSH_PATH, exist_ok=True)
+
+        with open(
+            os.path.join(SSH_PATH, '/gin_id_rsa'),
```
Comment (achilleas-k) [inline back-end/service.py]: This is probably why the whole key business wasn't working.  When one of the components in `os.path.join()` starts with a `/`, the previous parts are ignored, so this line produces `/gin_id_rsa`.

```suggestion
            os.path.join(SSH_PATH, 'gin_id_rsa'),
```

File: back-end/server.py [hunk 2]
```diff
@@ -8,59 +9,78 @@
...
 @app.route('/login', methods=['POST'])
 @cross_origin()
 def login():
     if request.method == "POST":
-        global auth
-        auth = authorize(username=request.json['username'], password=request.json['password'])
-        if not validUser(auth): abort(400)
-        print("User {} logged in".format(request.json['username']))
+
+        global username
```
Comment (achilleas-k) [inline back-end/server.py]: Using globals for the username and password isn't nice.  We can leave it for now, but make a note to change this.  You can change it to be a class with instance variables (properties) or just pass them back and forth between functions as needed.

File: back-end/server.py [hunk 3]
```diff
@@ -8,59 +9,78 @@
...
 
 
 @app.route('/')
 def index(): return render_template('index.html')
 
 
+@app.route('/logout', methods=['POST'])
+def logMeOut():
+
+        if request.method == "POST":
```
Comment (achilleas-k) [inline back-end/server.py]: Line is over-indented.

File: clone+annex [hunk 4]
```diff
@@ -1,2 +1,5 @@
 FROM ubuntu
-RUN apt-get update
\ No newline at end of file
+RUN apt-get update && apt-get install python3-pip openssh-server openssh-client -y
+RUN apt-get install git git-annex -y
+RUN pip3 install snakemake
+
```
Comment (achilleas-k) [inline clone+annex]: What is the purpose of this file?

File: back-end/service.py [hunk 5]
```diff
@@ -1,172 +1,301 @@
...
 
-	with tempfile.TemporaryDirectory() as temp_clone_path:
-		clone_path = clone(repo, auth.username, temp_clone_path)
-		designWorkflow(workflowFiles, clone_path)
-		designCIConfig(backPushFiles, annexFiles, clone_path)
-		push(clone_path, commitMessage)
-		clean(clone_path)
\ No newline at end of file
+def clean(path):
+    call(['rm', '-rf', path])
```
Comment (achilleas-k) [inline back-end/service.py]: Best not too rely on shell or OS specific commands when the language has it's own implementation.  See https://docs.python.org/3/library/shutil.html#shutil.rmtree

File: back-end/service.py [hunk 6]
```diff
@@ -1,172 +1,301 @@
...
+
+def ensureKeys(token):
+
+    SSH_PATH = os.path.join('/gin-proc', 'ssh')
+
+    if ensureKeysOnServer(token) and ensureKeysOnLocal(SSH_PATH):
+        return True
+
+    elif ensureKeysOnServer(token) and not ensureKeysOnLocal(SSH_PATH):
+        print("Key {} is installed on the server but not locally. \
```
Comment (mpsonntag) [inline back-end/service.py]: I guess a `.format` is missing at this print.

File: back-end/service.py [hunk 7]
```diff
@@ -1,172 +1,301 @@
...
+        Kindly delete your key online so we can install a fresh key pair.")
+
+        return False
+
+    elif not ensureKeysOnServer(token) and ensureKeysOnLocal(SSH_PATH):
+        print("Key {} is installed locally but not on the server.")
```
Comment (mpsonntag) [inline back-end/service.py]: Same `.format` missing as above.

File: back-end/service.py [hunk 8]
```diff
@@ -1,172 +1,301 @@
...
+                        input_files += "'{}' ".format(filename)
+                    lines.append('    - git annex get {}'.format(input_files))
+
+            elif '#add-notifications' in line:
+                addNotifications(lines, notifications)
+
+            else:
+                lines.append(line)
+
+    template.close()
```
Comment (mpsonntag) [inline back-end/service.py]: When using the `with` statement, explicitly closing the file afterwards is not necessary, the ContextManager takes care of that.

File: requirements.txt [hunk 9]
```diff
@@ -11,7 +11,10 @@ Flask-Cors==3.0.8
 future==0.17.1
 gitdb2==2.0.5
 GitPython==2.1.11
+<<<<<<< HEAD
```
Comment (mpsonntag) [inline requirements.txt]: Please remove the merge leftover.

File: clone+annex (hunk 4 above)
Comment (mrinalwahal) [reply clone+annex]: This image is used in `Drone.yml` configs for pipeline steps.

File: back-end/service.py (hunk 5 above)
Comment (mrinalwahal) [reply back-end/service.py]: Replaced with `shutil.rmtree`

File: back-end/service.py (hunk 6 above)
Comment (mrinalwahal) [reply back-end/service.py]: Cured. Won't exist in the next update.

File: requirements.txt (hunk 9 above)
Comment (mrinalwahal) [reply requirements.txt]: Done. Won't be there in the next update.

File: back-end/service.py (hunk 8 above)
Comment (mrinalwahal) [reply back-end/service.py]: Removed the closing tags with all `with` statements.

File: back-end/server.py (hunk 3 above)
Comment (mrinalwahal) [reply back-end/server.py]: Cured. Won't exist in the next update.

File: back-end/server.py (hunk 2 above)
Comment (mrinalwahal) [reply back-end/server.py]: Noted.

**Pull request merged**